pkg/agent/agent.go

```diff
@@ -0,0 +1,94 @@
+package agent
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/kubeapps/kubeapps/pkg/proxy"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/kube"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+)
+
+const driverEnvVar = "HELM_DRIVER"
+
+type Options struct {
+	ListLimit int
+	Timeout   int64
+}
+
+type Context struct {
+	ActionConfig *action.Configuration
+	AgentOptions Options
+}
+
+func ListReleases(context Context, namespace string, status string) ([]proxy.AppOverview, error) {
+	cmd := action.NewList(context.ActionConfig)
+	if namespace == "" {
+		cmd.AllNamespaces = true
+	}
+	cmd.Limit = context.AgentOptions.ListLimit
+	releases, err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+	appOverviews := make([]proxy.AppOverview, len(releases))
+	for i, r := range releases {
+		appOverviews[i] = appOverviewFromRelease(r)
+	}
+	return appOverviews, nil
+}
+
+func NewConfig(token, namespace string) *action.Configuration {
+	actionConfig := new(action.Configuration)
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+	config.BearerToken = token
+	config.BearerTokenFile = ""
+	clientset, err := kubernetes.NewForConfig(config)
+	store := createStorage(os.Getenv(driverEnvVar), namespace, clientset)
+	actionConfig.RESTClientGetter = nil
+	actionConfig.KubeClient = kube.New(nil)
+	actionConfig.Releases = store
+	actionConfig.Log = klog.Infof
```

Comment:
Out of interest, why do you want to store the logger in the action config?

Reply:
I don't know. [It's part of the `action.Configuration` type](https://github.com/helm/helm/blob/865c46c014cdb7622f97ae287ee92fb8a280f3a9/pkg/action/action.go#L84) from Helm 3.